pkg/token/cache.go

```diff
@@ -84,13 +83,13 @@ type LockedTokenCache struct {
 	podName string
 }
 
-func NewLockedTokenCache(tokenType string, idConfig *config.IdentityConfig) *LockedTokenCache {
+func NewLockedTokenCache(tokenType, namespace, podName string) *LockedTokenCache {
```

Comment:
Why would you not use the pointer?
- easy to read
- less parameters
- faster

Reply:
> Why would you not use the pointer?
- remove the dependency from `cache.go` to `config.go`, keep them decoupled
- easy to read => do not know which field will be used
- less parameters => should keep to interface to minimum (IdentityConfig is not 1 object, it implies all its fields)
- faster => agress

Reply:
> remove the dependency from cache.go to config.go, keep them decoupled

okay